sopel/bot.py

```diff
@@ -706,12 +706,8 @@ def call_rule(
         except PluginAbort as abort:
             LOGGER.warning("Plugin aborted while handling trigger %r", trigger, exc_info=abort)
 
-            if len(abort.args) == 1:
-                message = abort.args[0]
-                if not isinstance(message, str):
-                    message = repr(message)
-
-                self.say(message, trigger.sender)
+            if abort.message:
+                self.say(abort.message, trigger.sender)
```

Comment:
Now the only thing I have immediately is this idea that there should be a way to reply() instead. When I do the speak-and-return pattern this is intended to replace, I usually use reply.

Oh, and the subject of our IRC discussion just now, about how to control rate limiting behavior of the rule after this abort mechanism trips.

Reply:
Hmm, how would that work for events that weren't triggered by a channel message (i.e. something on an interval)?

I can see the case for making it reply-by-default where that is contextually meaningful, but it does seem like it would be a more elaborate implementation to get there.